app/controllers/kudos_controller.rb

```diff
@@ -66,6 +67,34 @@ def create
 
     if kudo.save
       render json: { kudo: kudo }, status: :created
+
+      # Success, send email notification
+      ReceivedKudosMailer.received_kudos_notification(kudo).deliver_now
+
+
+      # Notify user via slack. This is initalized in kudos/environment.rb.
+      # Can this be called once initially and then regularly
+      # updated via scheduled job?
+      slack_users_list = kudosbot.users_list.members
+      receiver_slack_name = ""
+
+      # Find slack user name for receiver. Should this be a helper method
+      # in ../helpers/application_helper? That method should take an
+      # array of IDs and return a hash of emails to slack usernames.
+      # i.e. helpers.find_slack_usernames([ID1, ID2]) =>
+      #   {ID1: slackname1, ID2, slackname2}
+
+      # Per Byron, we should probably just add slack usernames to
+      # the DB for all new users
+
+      slack_users_list.each do |user|
+        receiver_slack_name = user['name'] if user.profile["email"] == receiver_email
+      end
+      fail "No Slack user found with email #{receiver_email}" unless receiver_slack_name
+
+      # Post message to kudos channel in slack.
+      kudosbot.chat_postMessage(channel: '#kudos-dev', text: "Hey @#{receiver_slack_name}, you just received a kudos on http://kudos.factual.com!", as_user: true, "link_names": 1)
```

Comment:
I feel like all this Slack stuff should be broken off into two functions, so in `if kudo.save`, you can just call a private method (e.g. `send_slack_notification`) and in that function you can call something like `slack_username_from_user` to get the username (and throw an exception when it fails). Heck, that last function is probably general enough to be outside of this controller (either a helper or on the user model itself).

Also I have the same comment about hard-coding the kudos URL here — it should probably be passed into the docker as a configuration option.